src/lib/var.c

```diff
@@ -257,43 +268,53 @@ static char* prepare_search_key(char *name, unsigned int *name_length, const cha
 	return element;
 }
 
-static zval *get_arrayobject_storage(zval *parent, HashTable **properties, int *is_temp)
-{
 #if PHP_VERSION_ID >= 70400
+static zval *get_arrayobject_storage(zval *parent, HashTable **properties)
+{
+	*properties = zend_get_properties_for(parent, ZEND_PROP_PURPOSE_DEBUG);
+	return zend_hash_str_find(*properties, "\0ArrayObject\0storage", sizeof("*ArrayObject*storage") - 1);
+}
+
+static zval *get_splobjectstorage_storage(zval *parent, HashTable **properties)
+{
 	*properties = zend_get_properties_for(parent, ZEND_PROP_PURPOSE_DEBUG);
+	return zend_hash_str_find(*properties, "\0SplObjectStorage\0storage", sizeof("*SplObjectStorage*storage") - 1);
+}
+
+static zval *get_arrayiterator_storage(zval *parent, HashTable **properties)
+{
+	*properties = zend_get_properties_for(parent, ZEND_PROP_PURPOSE_DEBUG);
+	return zend_hash_str_find(*properties, "\0ArrayIterator\0storage", sizeof("*ArrayIterator*storage") - 1);
+}
 #else
+static zval *get_arrayobject_storage(zval *parent, HashTable **properties, int *is_temp)
+{
 	*properties = Z_OBJDEBUG_P(parent, *is_temp);
-#endif
 	return zend_hash_str_find(*properties, "\0ArrayObject\0storage", sizeof("*ArrayObject*storage") - 1);
 }
 
 static zval *get_splobjectstorage_storage(zval *parent, HashTable **properties, int *is_temp)
 {
-#if PHP_VERSION_ID >= 70400
-	*properties = zend_get_properties_for(parent, ZEND_PROP_PURPOSE_DEBUG);
-#else
 	*properties = Z_OBJDEBUG_P(parent, *is_temp);
-#endif
 	return zend_hash_str_find(*properties, "\0SplObjectStorage\0storage", sizeof("*SplObjectStorage*storage") - 1);
 }
 
 static zval *get_arrayiterator_storage(zval *parent, HashTable **properties, int *is_temp)
 {
-#if PHP_VERSION_ID >= 70400
-	*properties = zend_get_properties_for(parent, ZEND_PROP_PURPOSE_DEBUG);
-#else
 	*properties = Z_OBJDEBUG_P(parent, *is_temp);
-#endif
 	return zend_hash_str_find(*properties, "\0ArrayIterator\0storage", sizeof("*ArrayIterator*storage") - 1);
 }
+#endif
```

Comment:
I agree that separating it out into distinct blocks instead of an `ifdef` per function is much more readable.